sqlmesh/core/engine_adapter/spark.py

```diff
@@ -247,7 +247,10 @@ def _create_table_properties(
         if partitioned_by:
             partition_columns_property = exp.PartitionedByProperty(
                 this=exp.Schema(
-                    expressions=[exp.to_identifier(column) for column in partitioned_by]
+                    expressions=[
+                        column.this if isinstance(column, exp.Column) else column
```

Comment:
Should we fail if we detect a non-column expression?

Reply:
If only column expressions are valid, then yes. Please confirm.

Reply:
For vanilla spark - yes, I think so